supportedBackends/sfs/src/main/scala/cromwell/backend/impl/sfs/config/ConfigAsyncJobExecutionActor.scala

```diff
@@ -246,4 +250,50 @@ class DispatchedConfigAsyncJobExecutionActor(override val standardParams: Standa
   override def killArgs(job: StandardAsyncJob): SharedFileSystemCommand = {
     jobScriptArgs(job, "kill", KillTask)
   }
+
+  protected lazy val exitCodeTimeout: Option[Int] =
+    configurationDescriptor.backendConfig.as[Option[Int]](ExitCodeTimeoutConfig).map(math.abs)
+
+  override def pollStatus(handle: StandardAsyncPendingExecutionHandle): SharedFileSystemRunStatus = {
+    handle.previousStatus match {
+      case None =>
+        // Is not set yet the status will be set default to running
+        SharedFileSystemRunStatus("Running")
+      case Some(s) if (s.status == "Running" || s.status == "WaitingForReturnCode") && jobPaths.returnCode.exists =>
+        // If exitcode file does exists status will be set to Done always
+        SharedFileSystemRunStatus("Done")
+      case Some(s) if s.status == "Running" =>
+        // Exitcode file does not exist at this point, checking is jobs is still alive
+        if (exitCodeTimeout.isEmpty) s
```

Comment:
I would move this into the `case None` so that we jump straight to `WaitingForReturnCode ` if the timeout is not set.

Reply:
I tried this first but then I did hit a boxed error, this solution did work. Also don't know why that happened.

Reply:
btw, if you know a why this happens I'm happy to change this ofcourse

Reply:
I did mean *not* set - on reflection I suspect there are many *many* ways to structure this... TBH I don't think we'd gain much with my suggestion so I'm happy to leave it as-is for now